CMakeLists.txt

```diff
@@ -0,0 +1,104 @@
+cmake_minimum_required(VERSION 2.8)
+
+project(papas)
+set( papas_VERSION_MAJOR 0 )
+set( papas_VERSION_MINOR 1 )
+set( papas_VERSION_PATCH 0 )
+set( papas_VERSION "${papas_VERSION_MAJOR}.${papas_VERSION_MINOR}" )
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+
+if(APPLE)
+    set(CMAKE_SHARED_LIBRARY_SUFFIX ".so")
+    #tell mac to use RPATH
+
+endif(APPLE)
+
+#--- Declare options -----------------------------------------------------------
+option(papas_documentation "Whether or not to create doxygen doc target." ON)
+
+
+list(APPEND CMAKE_PREFIX_PATH $ENV{ROOTSYS} $ENV{PODIO} $ENV{FCCEDM})
+
+#--- Declare ROOT dependency ---------------------------------------------------
+if(APPLE)
+	set(CMAKE_MACOSX_RPATH 1)
+	set(CMAKE_OSX_DEPLOYMENT_TARGET 10.10)
+set(CMAKE_SHARED_LIBRARY_SUFFIX ".so")
+find_package(ROOT REQUIRED HINTS $ENV{ROOTSYS})
+add_definitions(${ROOT_CXX_FLAGS})
+include("${ROOT_USE_FILE}")
+else()
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+find_package(ROOT REQUIRED COMPONENTS RIO Tree)
+include_directories(${ROOT_INCLUDE_DIR})
+endif(APPLE)
+
+
+#---PODIO and FCC-EDM-----------------------------------------------------------
```

Comment:
The dependencies to fcc software should be optional. 
papas can work with a simple list of generated particles in input. 
e.g. a text file, a root tree, an HepMC event 
Depending on the software that is found we should provide various input back-ends: 
- HepMCReader
- RootTreeReader
- TextReader
- ?
we should discuss the design of these generic tools. For instance, the user could provide a text file with a custom format with the list of particles for each event.